### PR TITLE
Replaces uses of `layers.last()` by `_.last(layers.getDataLayers())`:

### LINKED ISSUE #4438: Table in "read-only" mode when creating new map with labels on top
**Steps:**

* Go to your dashboard
* Create a new empty map
* Come back to the dashboard, go to Datasets, and open the newly created dataset
* Once inside the dataset, try to double click the name to rename it

**Result:**

The table cannot be renamed because it's detected as "read-only" mode.

image

@alonsogarciapablo spotted the issue [here](https://github.com/CartoDB/cartodb/blob/master/lib/assets/javascripts/cartodb/old_common/header.js#L585)

### PR DESCRIPTION
**Pull request by alonsogarciapablo**
Fixes #4438. The last layer of the collection might now be a Tiled layer with labels and that was causing some issues when determining if a dataset/map was editable.

@xavijam PTAL. Thanks!

### CODE REVIEW

File: lib/assets/javascripts/cartodb/table/metadata_dialog.js [hunk 1]
```diff
@@ -136,12 +136,14 @@
...
-          return true
+          return true;
         } else {
-          return false
+          return false;
         }
       } else {
-        var table = this.vis.map.layers && this.vis.map.layers.last() &&  this.vis.map.layers.last().table;
+        var lastDataLayer = this.vis.map.layers &&
+          _.last(this.vis.map.layers.getDataLayers());
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/metadata_dialog.js]: Put them in the same line no? It is not really long.

File: lib/assets/javascripts/cartodb/table/metadata_dialog.js [hunk 2]
```diff
@@ -156,12 +158,15 @@
...
         if (this.model.permission.isOwner(this.user)) {
-          return true
+          return true;
         } else {
-          return false
+          return false;
         }
       } else {
-        var table = this.vis.map.layers && this.vis.map.layers.last().table;
+        var lastDataLayer = this.vis.map.layers &&
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/metadata_dialog.js]: same here

**Comment by xavijam** [comment]
Go ahead, makes sense!

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4951/))

File: lib/assets/javascripts/cartodb/table/metadata_dialog.js (hunk 1 above)
Comment (alonsogarciapablo) [inline lib/assets/javascripts/cartodb/table/metadata_dialog.js]: I personally find it long enough to break it into two lines. I can't read the whole line in my laptop for example :(

Comment (xavijam) [inline lib/assets/javascripts/cartodb/table/metadata_dialog.js]: What kind of resolution do you use!?!?!? :DDD.
Go ahead, no problem.

**Pull request merged**